### ISSUE TITLE
Propagate `loc="best"` down to AnchoredOffsetbox?

### ISSUE THREAD

**Issue by jklymak** [matplotlib/matplotlib#17946]
Folks often want a way to automatically place text, but not use a legend (#4489, #1313), but attempts to implement this have not passed muster (#12841)

As pointed out by @ImportanceofBeeingEarnest one way to make this more constrained is to use `AnchoredOffsetbox` which `legend` already uses, and just propagate the legend logic to the parent class.  The folks could put their text or other annotations in this box.  

> @andrzejnovak I think what you're asking for is rather a `loc="best"` option for [`AnchoredOffsetbox`](https://matplotlib.org/3.1.0/api/offsetbox_api.html#matplotlib.offsetbox.AnchoredOffsetbox), such that one could write `AnchoredText("my text", loc="best")`. This is very different from what this PR suggested, and most concerns raised here would not even apply.    

_Originally posted by @ImportanceOfBeingErnest in https://github.com/matplotlib/matplotlib/pull/12841#issuecomment-508057229_

**Comment by codeananda** [comment]
Could something like [this](https://stackoverflow.com/a/59109053/11829398) answer by [Chipmuenk](https://stackoverflow.com/users/2779500/chipmuenk) do the trick?

```python
import numpy as np
import matplotlib.pyplot as plt
import matplotlib.patches as mpl_patches

x = np.linspace(-1,1)
fig, ax = plt.subplots()
ax.plot(x, x*x)

# create a list with two empty handles (or more if needed)
handles = [mpl_patches.Rectangle((0, 0), 1, 1, fc="white", ec="white", 
                                 lw=0, alpha=0)] * 2

# create the corresponding number of labels (= the text you want to display)
labels = []
labels.append("pi = {0:.4g}".format(np.pi))
labels.append("root(2) = {0:.4g}".format(np.sqrt(2)))

# create the legend, supressing the blank space of the empty line symbol and the
# padding between symbol and label by setting handlelenght and handletextpad
ax.legend(handles, labels, loc='best', fontsize='small', 
          fancybox=True, framealpha=0.7, 
          handlelength=0, handletextpad=0)
plt.show()
```

<img width="486" alt="image" src="https://github.com/matplotlib/matplotlib/assets/51246969/9304d4a4-a541-4917-b703-99c72f7b74e1">

**Comment by timhoffm** [comment]
> Could something like [this](https://stackoverflow.com/a/59109053/11829398) answer by [Chipmuenk](https://stackoverflow.com/users/2779500/chipmuenk) do the trick?

Abusing legend as a simple list of texts is a hacking workaround. It may solve the problem for a certain subset of cases if users really need this capability in current versions of matplotlib. However, it cannot be an officially recommended way of working.

**Comment by codeananda** [comment]
Yes it is a hack. But it's the only one that gives you the option to use `loc='best'`, right?

**Comment by timhoffm** [comment]
> But it's the only one that gives you the option to use `loc='best'`, right? 

Yes, as long as this issue is not implemented it's the only option. But it's not the general solution needed to implement this issue.
